test/cases/ddl/table_partition.test

```diff
@@ -0,0 +1,105 @@
+CREATE TABLE tp5 (col1 INT, col2 CHAR(5), col3 DATE) PARTITION BY LINEAR KEY ALGORITHM = 1 (col3) PARTITIONS 5;
+CREATE TABLE tp6 (col1 INT, col2 CHAR(5), col3 DATE) PARTITION BY KEY(col1, col2) PARTITIONS 4;
+CREATE TABLE tp7 (col1 INT NOT NULL PRIMARY KEY, col2 DATE NOT NULL, col3 INT NOT NULL, col4 INT NOT NULL) PARTITION BY KEY(col1) PARTITIONS 4;
+CREATE TABLE tp8 (col1 INT, col2 CHAR(5)) PARTITION BY HASH(col1);
+CREATE TABLE tp9 (col1 INT, col2 CHAR(5)) PARTITION BY HASH(col1) PARTITIONS 4;
+CREATE TABLE tp10 (col1 INT, col2 CHAR(5), col3 DATETIME) PARTITION BY HASH (YEAR(col3));
+CREATE TABLE tp11 (col1 INT, col2 CHAR(5), col3 DATE) PARTITION BY LINEAR HASH( YEAR(col3)) PARTITIONS 6;
+CREATE TABLE tp12 (col1 INT NOT NULL PRIMARY KEY, col2 DATE NOT NULL, col3 INT NOT NULL, col4 INT NOT NULL) PARTITION BY HASH(col1) PARTITIONS 4;
+CREATE TABLE tp13 (
+    id INT NOT NULL PRIMARY KEY,
+    fname VARCHAR(30),
+    lname VARCHAR(30),
+    hired DATE NOT NULL DEFAULT '1970-01-01',
+    separated DATE NOT NULL DEFAULT '9999-12-31',
+    job_code INT NOT NULL,
+    store_id INT NOT NULL
+)
+PARTITION BY RANGE (id) (
+    PARTITION p0 VALUES LESS THAN (6),
+    PARTITION p1 VALUES LESS THAN (11),
+    PARTITION p2 VALUES LESS THAN (16),
+    PARTITION p3 VALUES LESS THAN (21)
+);
+
+
+CREATE TABLE tp14 (
+	id INT NOT NULL,
+	fname VARCHAR(30),
+	lname VARCHAR(30),
+	hired DATE NOT NULL DEFAULT '1970-01-01',
+	separated DATE NOT NULL DEFAULT '9999-12-31',
+	job_code INT,
+	store_id INT
+)
+PARTITION BY RANGE ( YEAR(separated) ) (
+	PARTITION p0 VALUES LESS THAN (1991),
+	PARTITION p1 VALUES LESS THAN (1996),
+	PARTITION p2 VALUES LESS THAN (2001),
+	PARTITION p3 VALUES LESS THAN MAXVALUE
+);
+CREATE TABLE tp15 (
+	a INT NOT NULL,
+	b INT NOT NULL
+)
+PARTITION BY RANGE COLUMNS(a,b) PARTITIONS 4 (
+	PARTITION p0 VALUES LESS THAN (10,5),
+	PARTITION p1 VALUES LESS THAN (20,10),
+	PARTITION p2 VALUES LESS THAN (50,20),
+	PARTITION p3 VALUES LESS THAN (65,30)
+);
+CREATE TABLE tp16 (
+	id   INT PRIMARY KEY,
+	name VARCHAR(35),
+	age INT unsigned
+)
+PARTITION BY LIST (id) (
+	PARTITION r0 VALUES IN (1, 5, 9, 13, 17, 21),
+	PARTITION r1 VALUES IN (2, 6, 10, 14, 18, 22),
+	PARTITION r2 VALUES IN (3, 7, 11, 15, 19, 23),
+	PARTITION r3 VALUES IN (4, 8, 12, 16, 20, 24)
+);
+CREATE TABLE tp17 (
+	id   INT,
+	name VARCHAR(35),
+	age INT unsigned
+)
+PARTITION BY LIST (id) (
+	PARTITION r0 VALUES IN (1, 5, 9, 13, 17, 21),
+	PARTITION r1 VALUES IN (2, 6, 10, 14, 18, 22),
+	PARTITION r2 VALUES IN (3, 7, 11, 15, 19, 23),
+	PARTITION r3 VALUES IN (4, 8, 12, 16, 20, 24)
+);
+CREATE TABLE tp18 (
+	a INT NULL,
+	b INT NULL
+)
+PARTITION BY LIST COLUMNS(a,b) (
+	PARTITION p0 VALUES IN( (0,0), (NULL,NULL) ),
+	PARTITION p1 VALUES IN( (0,1), (0,2), (0,3), (1,1), (1,2) ),
+	PARTITION p2 VALUES IN( (1,0), (2,0), (2,1), (3,0), (3,1) ),
+	PARTITION p3 VALUES IN( (1,3), (2,2), (2,3), (3,2), (3,3) )
+);
+show tables;
```

Comment:
There need to have a show tables that prints out all columns, etc.   We need to at least print out some partition by information.

Reply:
Yes, a metadata table needs to be added later to record partition information

Reply:
So you plan to do it later?   You must update issue, document there is unfinished work.

Reply:
OK，I will do it.

Reply:
I have created a subtask to record and handle this case，issue #4749